lib/pbench/server/api/resources/query_apis/table_of_content.py

```diff
@@ -76,14 +77,15 @@ def postprocess(self, es_json: JSON, context: CONTEXT) -> JSON:
                 if parent != directory:   # This condition is especially when parent and directory = '/'
                     t = toc["_source"]["name"]
                     dir_list.append(t)
-                if parent == directory or parent == direct:
-                    file_count = len(toc["_source"]["files"])
+                if not file_list:
+                    if parent == directory or parent == direct:
+                        file_list = toc["_source"]["files"]
```

Comment:
And, of course, it should go without saying, but I'll say it anyway....  The API design can (and probably _should_) be independent of the underlying implementation and database design.  So, if we can decide exactly what we want the API to do, we can implement that today as best we can, and, in the future, if it proves insufficient, we can change the underlying implementation, including replacing the database with a different format, and we can do so without disrupting our clients.

So, really, we should decide how we want the interface to operate, and then put together an initial implementation; we can then iterate the implementation as needed, including things like reindexing the data, if that's warranted.

Reply:
I am able to fetch queries without any fail using 'filters' instead of 'must'. All thanks to @webbnh, I wanted to clear out the air by asking this question, we are moving towards progressive disclosure or we have to disclose the next sub-directory as well?

client clicks on: `\1-IT1\` directory the response should be next list of directories(for-example: `\1-IT1\sample-1`) and their sub-directories(for -ex: `\1-IT1\sample-1\abc`) 
`parent: \1-IT1\`
`directory: \1-IT1\sample-1`
`sub-directory: \1-IT1\sample-1\abc`

or

client clicks on: `\1-IT1\` directory the response should be next list of directories(for-example: `\1-IT1\sample-1`)
`parent: \1-IT1\`
`directory: \1-IT1\sample-1`

Reply:
There's not much point in returning an additional level of subdirectories when the client will have to make another call for each in order to get the *files* anyway. I think that calling with `/1_IT1` should return the direct children (regular files, symlinks, and directories) of `/1_IT1` and nothing else.

Using Webb's fairly convoluted disjunction, that's what we're getting. On the dataset I was examining, for example, there's a single file in `/1-IT1`, `reference-result` and a single subdirectory, `/1-IT1/sample1` so with a query for `"/1-IT1"` you'd get something like this:

```
{
  "files": [
    {
      "name": "reference-result",
      "mtime": "2021-09-21T12:40:50",
      "size": 0,
      "mode": "0o777",
      "type": "sym",
      "linkpath": "sample1"
     }
  ],
  "directories": [
    {
      "name": "sample1",
      "mtime": "2021-09-21T12:39:49",
      "mode": "0o755"
    }
  ]
}
```

Although I also like the idea of returning a single list with `"type": "dir"` for directories, we do expect the directories to be treated separately (e.g., displayed with a disclosure triangle to cause fetching the next level), so separating them probably makes sense.